File: serve.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"os"
	"os/exec"
	"path/filepath"
	"runtime"
)

type ServerInfo struct {
	Language string `json:"language"`
	Command  string `json:"command"`
	BaseDir  string `json:"baseDir"`
}

func main() {
	port := "8080"
	_, filename, _, _ := runtime.Caller(0)
	baseDir := filepath.Dir(filename)
	projectRootDir := filepath.Join(baseDir, "..")

	http.HandleFunc("/server-info", func(w http.ResponseWriter, r *http.Request) {
		info := ServerInfo{
			Language: "Go",
			Command:  "go run serve.go",
			BaseDir:  baseDir,
		}
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(info)
	})

	http.HandleFunc("/kanban.json", func(w http.ResponseWriter, r *http.Request) {
		localPath := filepath.Join(baseDir, "kanban.json")
		rootPath := filepath.Join(projectRootDir, "kanban.json")

		target := ""
		if _, err := os.Stat(localPath); err == nil {
			target = localPath
		} else if _, err := os.Stat(rootPath); err == nil {
			target = rootPath
		}

		if target != "" {
			w.Header().Set("Content-Type", "application/json")
			http.ServeFile(w, r, target)
		} else {
			http.Error(w, "kanban.json not found", http.StatusNotFound)
		}
	})

	// Serve UI files
	http.Handle("/", http.FileServer(http.Dir(baseDir)))

	url := fmt.Sprintf("http://localhost:%s/viewer.html", port)
	fmt.Printf("🚀 AI Command Center starting at %s\n", url)
	fmt.Printf("Serving from: %s\n", baseDir)
	
	go openBrowser(url)

	err := http.ListenAndServe(":"+port, nil)
	if err != nil {
		fmt.Printf("❌ Error: %v\n", err)
	}
}

func openBrowser(url string) {
	var err error
	switch runtime.GOOS {
	case "linux":
		err = exec.Command("xdg-open", url).Start()
	case "windows":
		err = exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
	case "darwin":
		err = exec.Command("open", url).Start()
	}
	if err != nil {
		fmt.Printf("⚠️ Browser error: %v\n", err)
	}
}

File: serve.js
const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

const PORT = 8080;
const url = `http://localhost:${PORT}/viewer.html`;
const baseDir = __dirname;
const projectRootDir = path.join(baseDir, '..');

const mimeTypes = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpg',
    '.gif': 'image/gif',
};

const server = http.createServer((req, res) => {
    // 1. Serve Server Info
    if (req.url === '/server-info') {
        res.writeHead(200, { 'Content-Type': 'application/json' });
        return res.end(JSON.stringify({ 
            language: 'Node.js', 
            command: 'node serve.js',
            baseDir: baseDir 
        }));
    }

    let urlPath = req.url.split('?')[0];
    if (urlPath === '/' || urlPath === '') urlPath = '/viewer.html';

    // 2. Specialized logic for kanban.json (Check local then parent)
    if (urlPath === '/kanban.json') {
        const localPath = path.join(baseDir, 'kanban.json');
        const rootPath = path.join(projectRootDir, 'kanban.json');
        
        if (fs.existsSync(localPath)) {
            res.writeHead(200, { 'Content-Type': 'application/json' });
            return fs.createReadStream(localPath).pipe(res);
        } else if (fs.existsSync(rootPath)) {
            res.writeHead(200, { 'Content-Type': 'application/json' });
            return fs.createReadStream(rootPath).pipe(res);
        } else {
            res.writeHead(404);
            return res.end('kanban.json not found in skill folder or project root.');
        }
    }

    // 3. Serve other files from skill folder
    const filePath = path.join(baseDir, urlPath);
    const extname = String(path.extname(filePath)).toLowerCase();
    const contentType = mimeTypes[extname] || 'application/octet-stream';

    fs.readFile(filePath, (error, content) => {
        if (error) {
            res.writeHead(error.code === 'ENOENT' ? 404 : 500);
            res.end(error.code === 'ENOENT' ? 'File not found: ' + urlPath : 'Server error');
        } else {
            res.writeHead(200, { 'Content-Type': contentType });
            res.end(content, 'utf-8');
        }
    });
});

server.listen(PORT, () => {
    console.log(`🚀 AI Command Center starting at ${url}`);
    console.log(`Serving from: ${baseDir}`);
    console.log('Press Ctrl+C to stop.');

    const start = (process.platform === 'darwin' ? 'open' : process.platform === 'win32' ? 'start' : 'xdg-open');
    exec(`${start} ${url}`);
});
